### PR TITLE
[BTFS-826] & [BTFS-816] 

### PR DESCRIPTION
**Pull request by wxue**
* Add config for periodic refresh -- 60 mins as default.
* Save/update local datastore
* move both hosts sync and analytics sync in spin package

* Feature change: missing --dc flag should get Experimental.Analytics setting from config file

Related PR: [go-btfs-config PR 12](https://github.com/TRON-US/go-btfs-config/pull/12)
Released in: https://github.com/TRON-US/go-btfs-config/releases/tag/v0.1.7

### CODE REVIEW

**Comment by taiyangc** [comment]
Please link the go-btfs-config PR in the description

File: spin/hosts.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+const (
+	syncPeriod = 60 * time.Minute
+)
+
+func Hosts(node *core.IpfsNode) {
+	configuration, err := node.Repo.Config()
+	if err != nil {
+		fmt.Printf("Failed to get configuration %s\n", err)
```
Comment (taiyangc) [inline spin/hosts.go]: should log.Error

File: spin/hosts.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	}
+}
+
+func perodicSyncHosts(node *core.IpfsNode, mode string) {
+	tick := time.NewTicker(syncPeriod)
+	defer tick.Stop()
+	for range tick.C {
+		err := commands.SyncHosts(node, mode)
+		if err != nil {
+			fmt.Println("Failed to sync hosts: ", err)
```
Comment (taiyangc) [inline spin/hosts.go]: log

**Comment by wxue** [comment]
@taiyangc comments resolved, tests got passed. ready for another review when you have time, thanks

**Pull request merged**